rq/worker.py

```diff
@@ -950,12 +950,33 @@ def run_maintenance_tasks(self):
         self.clean_registries()
         Group.clean_registries(connection=self.connection)
 
+    def _pubsub_exception_handler(self, exc: Exception, pubsub: "PubSub", pubsub_thread: "PubSubWorkerThread") -> None:
+        """
+        This exception handler allows the pubsub_thread to continue & retry to
+        connect after a connection problem the same way the main worker loop
+        indefinitely retries.
+        redis-py internal mechanism will restore the channels subscriptions
+        once the connection is re-established.
+        """
+        if isinstance(exc, (redis.exceptions.ConnectionError)):
+            self.log.error(
+                "Could not connect to Redis instance: %s Retrying in %d seconds...",
+                exc,
+                2,
+            )
+            print("Connection Error, waiting 2 seconds before retry")
+            time.sleep(2.0)
+        else:
```

Comment:
For completeness' sake, is there an easy way to cover this line in test?

Reply:
@selwin new commit should be fine.